### PR TITLE
Migrate to use GuCDK

### PR DESCRIPTION
**Pull request by jonflynng**
## What does this change?

Migrates from vanilla cloudformation to `GuCDK`. The migration maintains the same functionality but provides improved security, maintainability, and type safety through TypeScript. Future updates to `GuCDK` means reduced maintenance overhead and ensures we stay current with security practices.

We followed the migration docs [here](https://github.com/guardian/cdk/blob/main/docs/migration-guide-ec2.md).

## How to test

We have deployed this to PROD and tested it at https://status.ophan.co.uk/.

## How can we measure success?

Migrate successfully to GuCDK while retaining the same behaviour of the app.

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: cdk/lib/status-app.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			path: '/management/healthcheck',
+			healthyThresholdCount: 2,
+			unhealthyThresholdCount: 2,
+			interval: Duration.seconds(10),
+			timeout: Duration.seconds(5),
+		};
+
+		Tags.of(ec2.autoScalingGroup).add('SystemdUnit', `${stack}.service`);
+		Tags.of(ec2.autoScalingGroup).add('Management', 'port=9000');
+		Tags.of(ec2.autoScalingGroup).add('Role', `${stack}-status-app`);
```
Comment (SHession) [inline cdk/lib/status-app.ts:75]: What do the `management` and `role` tags do?

**Review by SHession — COMMENTED** [review]
Looks like a good start, it would be worth adding a description to this PR. 

I've added a few comments for you to consider.

File: cdk/lib/status-app.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+			},
+			tableName: 'StatusAppConfig',
+			partitionKey: { name: 'key', type: AttributeType.STRING },
+			billingMode: BillingMode.PROVISIONED,
+			readCapacity: 1,
+			writeCapacity: 1,
+		});
+
+		if (hostedZone.valueAsString) {
+			new GuCname(this, 'MainDnsEntry', {
```
[the remark this answers is not in the document]
Comment (jonflynng) [reply cdk/lib/status-app.ts:89]: Ah yep, you're right

File: cdk/lib/status-app.ts (hunk 1 above)
Comment (jonflynng) [reply cdk/lib/status-app.ts:75]: Will check, they were in original cfn but may not be needed now

File: cdk/bin/cdk.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import 'source-map-support/register';
+import { GuRoot } from '@guardian/cdk/lib/constructs/root';
+import { StatusApp } from '../lib/status-app';
+
+const app = new GuRoot();
+new StatusApp(app, 'StatusApp-euwest-1-PROD', {
```
[the remark this answers is not in the document]
Comment (jonflynng) [reply cdk/bin/cdk.ts:6]: this was auto-generated from the initial gucdk commands we ran, can remove though

File: cdk/lib/status-app.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+			unhealthyThresholdCount: 2,
+			interval: Duration.seconds(10),
+			timeout: Duration.seconds(5),
+		};
+
+		Tags.of(ec2.autoScalingGroup).add('SystemdUnit', `${stack}.service`);
+		Tags.of(ec2.autoScalingGroup).add('Management', 'port=9000');
+		Tags.of(ec2.autoScalingGroup).add('Role', `${stack}-status-app`);
+
+		new GuDynamoTable(this, 'ConfigTable', {
```
[the remark this answers is not in the document]
Comment (SHession) [reply cdk/lib/status-app.ts:77]: I would expect to have a separate param per property.

File: cdk/lib/status-app.ts (hunk 1 above)
Comment (jonflynng) [reply cdk/lib/status-app.ts:75]: I'm not seeing them when searching the GuCDK library. We could remove them and test if we think they're unnecessary.

File: cdk/lib/status-app.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+			devXBackups: {
+				enabled: true,
+			},
+			tableName: 'StatusAppConfig',
```
Comment (SHession) [inline cdk/lib/status-app.ts:82]: This name will prevent there from being more than a single stack in place

File: cdk/lib/status-app.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+			partitionKey: { name: 'key', type: AttributeType.STRING },
+			billingMode: BillingMode.PROVISIONED,
+			readCapacity: 1,
+			writeCapacity: 1,
+		});
+
+		if (hostedZone.valueAsString) {
+			new CfnRecordSet(this, "foo", {
```
Comment (SHession) [inline cdk/lib/status-app.ts:82]: Can we give this a sensible name:

```suggestion
			new CfnRecordSet(this, "CNameRecord", {
```

**Review by SHession — APPROVED** [review]
Given the amount of work this involved, it would be good to have some more detail in the PR description, especially any things which caught us out that we might encounter again. 

Otherwise, the changes are [working in PROD](https://status.ophan.co.uk/PROD) and LGTM

File: cdk/.DS_Store
Comment (SHession) [inline cdk/.DS_Store:1]: Should this file be ignored?

Comment (jonflynng) [reply cdk/.DS_Store:1]: It should

**Pull request merged**